components/autoAlign.py

```diff
@@ -50,50 +54,69 @@ def toggleShootAfterComplete(self):
         else:
             self.shootAfterComplete = True
         return self.shootAfterComplete
+    #Stops robot from running until starting is true
+    @state
+    def idling(self):
+        if self.starting:
+            self.starting = False
+            log.error("starting")
+            self.next_state("start")
+        else:
+            self.next_state("idling")
 
     @state(first=True)
     def start(self):
         # If limelight can see something
-        self.tx = self.limeTable.getNumber("tx", -50)
-        if self.tx != -50 or self.tx != 0:
+        self.DeviationX = self.limeTable.getNumber("tx", -50)
+        if self.DeviationX != -50 or self.DeviationX != 0:
             # "-50" is the default value, so if that is returned,
             # nothing should be done because there is no connection.
-            tx = self.limeTable.getNumber("tx", -50)
-            if tx != -50 and tx != 0:
-                if tx > self.maxAimOffset:
-                    if tx < self.PIDAimOffset:
-                        self.speed = self.calc_PID(tx)
-                    else:
-                        self.speed = self.DumbSpeed
-                    self.next_state_now("adjust_self_right")
-
-                elif tx < -1 * self.maxAimOffset:
-                    if tx > -1 * self.PIDAimOffset:
-                        self.speed = -1 * self.calc_PID(tx)
+            values = [
+                     [[self.maxAimOffset, self.PIDAimOffset],self.DumbSpeed],
+                     [[self.PIDAimOffset,"End"],self.DumbSpeed]
+                     ]
+
+            """
+            If DeviationX value is in between the minimum and maximum values
+            then the speed is set to the second array. if only one value needs to be check put
+            "End" as max value.
+            [self.min, self.max],[speed]
+            """
+
+            self.speed = 0
+            self.AbsoluteX = abs(self.DeviationX)
+            for dists, speed in values:
+
+                if (dists[1] == "End"
+                    or len(dists) == 2
+                    and dists[0] < self.AbsoluteX
+                    and self.AbsoluteX < dists[1]):
+                    if speed == "PID":
+                        self.speed = self.calc_PID(self.DeviationX)
                     else:
-                        self.speed = self.DumbSpeed
-                    self.next_state_now("adjust_self_left")
+                        self.speed = speed
+                    self.next_state("adjust_self")
```

Comment:
This causes an infinite loop (which we stop in robot.py, but it's better if the component exits by itself)